dev_tools/lib/solidus/release_drafter/client.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'octokit'
+require 'faraday'
+require_relative 'draft'
+
+module Solidus
+  class ReleaseDrafter
+    # Octokit wrapper for the network calls we need to work with GH releases.
+    #
+    # @api private
+    class Client
+      def initialize(github_token:, repository:)
+        @repository = repository
+        @client = Octokit::Client.new(access_token: github_token).tap { |c| c.auto_paginate }
+      end
+
+      def fetch_draft(tag:)
+        release = @client.releases(@repository).find do |release|
+          release.name == tag
+        end
+        release ? Draft.new(url: release.url, content: release.body) : Draft.empty
+      end
+
+      def fetch_pr(pr_number:)
+        @client.pull(@repository, pr_number)
+      end
+
+      def create_draft(draft:, tag:, branch:)
+        @client.create_release(
+          @repository,
+          tag,
+          name: tag,
+          target_commitish: branch,
+          body: draft.content,
+          draft: true
+        ) && draft
+      end
+
+      def update_draft(draft:, tag:, branch:)
+        @client.update_release(
+          draft.url,
+          name: tag,
+          body: draft.content,
+          draft: true,
+          tag_name: tag,
+          target_commitish: branch
+        ) && draft
+      end
```

Comment:
Not sure I get it. The `Solidus::ReleaseDrafter::Client` class wraps both all the external requests and the Octokit API. On top of that, it works with our custom `Draft` data structure. We're also defining other methods, like `#fetch_pr` or `#fetch_draft`.

Reply:
I could have misread it, the impression I got was that it could have inlined everywhere it's used since each method is only used in one place and the wrapping is rather thin. But as I said I have no issue in keeping it as it is 🙌